app/services/classroom_config.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class ClassroomConfig
+  CONFIGURABLES = %w[issues].freeze
+
+  attr_reader :github_repository
+
+  def initialize(github_repository)
+    raise ArgumentError, "Invalid configuration repo" unless github_repository.branch_present? "github-classroom"
+    @github_repository = github_repository
+  end
+
+  def setup_repository(repo)
+    configs_tree = @github_repository.branch_tree("github-classroom")
+    configs_tree.tree.each do |config|
+      send("generate_#{config.path}", repo, config.sha) if CONFIGURABLES.include? config.path
```

Comment:
I usually get nervous when we use `send` but it looks like this is only from GitHub API responses correct?